### ISSUE TITLE
GradCAM not generated for some architectures

### ISSUE THREAD

**Issue by rodrigocmichelassi** [victor-nasc/SimpleFewShot#9]
When trying to generate GradCAM explainability for some architectures, no heatmap is generated. Some of the architectures are R2D2, DSN and MSENet. 

My thoughts is that this happen to MSENet due to the attention layers, but I'm not sure. For R2D2 and DSN it seems to be an network architecture problem, since it can not backpropagate gradients.

PS: Architectures, such as MetaOptNet and MTL, are able to generate GradCAM with no issues.

Replicate: 

Train a model (2-way 5-shot, for example) and save its checkpoints. Now, run the GradCAM generation:

```bash
python explain.py --model "msenet" --weights "msenet2w5s.pth" --ways 2 --shots 5 --gpu 0 --image "img00990" --target-class "hemorrhage" --classes "hemorrhage,healthy"
```

Expected behavior:
- A heatmap with activations around the are that contains the main feature for the image classification.

Actual behavior: 
- The resulting image is blue, but has no appearing activation whatsoever. It's a heatmap with no heat points.
